processor.go

```diff
@@ -442,6 +442,123 @@ func callInterfaceFunc(handler, input interface{}, streaming bool) (intf interfa
 	return
 }
 
+// ProcessClientStreamRequest allows clients to push multiple messages
+// asynchronously to the same service with the same connection. Unlike in
+// ProcessClientRequest, we take a channel of inputs that can be filled and will
+// subsequently call the service with any new messages received in the channel.
+func (p *ServiceProcessor) ProcessClientStreamRequest(req *http.Request, path string,
+	clientInputs chan []byte) ([]byte, *StreamingTunnel, error) {
+
+	outChan := make(chan []byte, 100)
+	buf := <-clientInputs
+	mh, ok := p.handlers[path]
+
+	reply, stopServiceChan, err := func() (interface{}, chan bool, error) {
```

Comment:
Is there an actual need/benefit of using a function here ?

Reply:
except if we need the code to be harder to read I guess no ;)